src/fmgc/src/flightplanning/LegsProcedure.ts

```diff
@@ -200,18 +201,26 @@ export class LegsProcedure {
               }
 
               if (mappedLeg !== undefined) {
-                  mappedLeg.legAltitudeDescription = currentLeg.altDesc;
+                  if (this.approachType === ApproachType.APPROACH_TYPE_ILS && (currentLeg.fixTypeFlags & FixTypeFlags.FAF) > 0) {
+                      if (currentLeg.altDesc === AltitudeDescriptor.At) {
+                          mappedLeg.legAltitudeDescription = AltitudeDescriptor.G;
+                      } else {
+                          mappedLeg.legAltitudeDescription = AltitudeDescriptor.H;
+                      }
+                  } else {
+                      mappedLeg.legAltitudeDescription = currentLeg.altDesc;
+                  }
                   mappedLeg.legAltitude1 = currentLeg.altitude1 * 3.28084;
                   mappedLeg.legAltitude2 = currentLeg.altitude2 * 3.28084;
                   mappedLeg.speedConstraint = currentLeg.speedRestriction;
                   mappedLeg.turnDirection = currentLeg.turnDirection;
                   mappedLeg.additionalData.legType = currentLeg.type;
                   mappedLeg.additionalData.overfly = currentLeg.flyOver;
+                  mappedLeg.additionalData.fixTypeFlags = currentLeg.fixTypeFlags;
 
                   mappedLeg.additionalData.distance = currentLeg.distanceMinutes ? undefined : currentLeg.distance / 1852;
                   mappedLeg.additionalData.distanceInMinutes = currentLeg.distanceMinutes ? currentLeg.distance : undefined;
                   mappedLeg.additionalData.course = currentLeg.trueDegrees ? currentLeg.course : A32NX_Util.magneticToTrue(currentLeg.course, Facilities.getMagVar(mappedLeg.infos.coordinates.lat, mappedLeg.infos.coordinates.long));
-                  mappedLeg.additionalData.overfly = currentLeg.flyOver;
```

Comment:
Out of curiosity, why is this change required?

Reply:
Duplicate... it's already done a few lines above. 😆